lib/components/custom_grid_view.dart

```diff
@@ -60,19 +61,20 @@ class GridRow extends StatelessWidget {
 
   List<Widget> get spacedChildren {
     final newChildren = <Widget>[];
-    for (var child in children) {
+
+    var missingColumns = columnCount - children.length;
+    final childrenFilled = [
+      ...children,
+      ...List.generate(missingColumns, (_) => const SizedBox.shrink()),
```

Comment:
doesnt removing the Spacer() stuff mess up the width?
I thought the point of filling this array was to make flutter calculate the width as if there were _columnCount_ elements.

Reply:
Maybe we can just simplify with a simple LayoutBuilder() wrapped around the row, which would give us the maxWidth.
Then like 
```dart
children.map((child) => Padding(padding: last ? gap : 0, SizedBox(width: maxWidth / columnCount, child)))
```

Reply:
did we ever consider [GridView](https://api.flutter.dev/flutter/widgets/GridView-class.html)?

Reply:
> doesnt removing the Spacer() stuff mess up the width? I thought the point of filling this array was to make flutter calculate the width as if there were _columnCount_ elements.

All the childrenFilled elements are wrapped in Expanded at line 72. Used SizedBox instead of Spacer because Spacer cannot be a child of Expanded.

Reply:
> Maybe we can just simplify with a simple LayoutBuilder() wrapped around the row, which would give us the maxWidth. Then like
> 
> ```dart
> children.map((child) => Padding(padding: last ? gap : 0, SizedBox(width: maxWidth / columnCount, child)))
> ```

I agree that this makes the code simpler. I think I didn't use it before to avoid manual calculation.

Reply:
Ah, of course. what do you think about my suggestions above?

Reply:
I can open a new task for refactoring this

Reply:
https://github.com/bcc-code/bcc-media-app/issues/197

Reply:
> did we ever consider [GridView](https://api.flutter.dev/flutter/widgets/GridView-class.html)?

Yes. The problem with GridView is that we can only specify the aspect ratio and not width or height.